### ISSUE TITLE
Running with docker run reports no errors, with cli runs way too long

### ISSUE THREAD

**Issue by daften** [codeclimate/codeclimate#533]
If I run the following command, I get the output shown.
```
docker run \
  --interactive --tty --rm \
  --env CODECLIMATE_CODE="/code" \
  --volume "$PWD":/code \
  --volume /var/run/docker.sock:/var/run/docker.sock \
  --volume /tmp/cc:/tmp/cc \
  codeclimate/codeclimate analyze .
Starting analysis
Running phpcodesniffer: Done!
Running markdownlint: Done!

Analysis complete! Found 0 issues.
```

Even if I put errors in the code on purpose, 0 issues.

If I run with the CLI tool with `codeclimate analyze .` it runs for ages, I abort it because it uses up CPU cycles like it's trying to roast a pig on my laptop.

If I run with `codeclimate analyze` I get the correct output.

```
codeclimate analyze
Starting analysis
Running phpcodesniffer: Done!
Running phpmd: Done!
Running duplication: Done!
Running fixme: Done!
Running markdownlint: Done!

== .idea/workspace.xml (1 issue) ==
...

Analysis complete! Found 33 issues.
```

I first ran with the . to indicate the folder, because that's what the command help actually says, there should be a path argument. That is apparently wrong, although I don't understand why right now. And why it runs for so long, not a clue.

The fact that running it with docker:run doesn't give results is even stranger, because I followed the instructions from the README file.

I haven't included the debug output yet, because I'd first like feedback  - and because I only get this when using it with `codeclimate analyze .` atm
```
Starting analysis
[DEBUG] phpcodesniffer:stable engine config: {"enabled":true,"config":{"standard":"PSR2"},"include_paths":["./"]}
[DEBUG] docker run: ["docker", "run", "--name", "cc-engines-phpcodesniffer-stable-d19affa7-ea5d-47c2-b4d4-91cdbb58af7e", "--cap-drop", "all", "--label", "com.codeclimate.label=977c5fee-2703-4bc3-8d1e-6fb0bb90171e", "--memory", "512000000", "--memory-swap", "-1", "--net", "none", "--rm", "--volume", "/Users/blommedi/Workspaces/CI/robo-drupal-console:/code:ro", "--volume", "/tmp/cc/8eeb1c5b-cdde-4339-bc03-ee8c448454a7:/config.json:ro", "--user", "9000:9000", "codeclimate/codeclimate-phpcodesniffer"]
```

I use Mac, installed the CLI tool through brew after brew update.

  ```
Darwin nbdig36445.gentgrp.gent.be 15.6.0 Darwin Kernel Version 15.6.0: Thu Sep  1 15:01:16 PDT 2016; root:xnu-3248.60.11~2/RELEASE_X86_64 x86_64
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 23:26:11 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 23:26:11 2016
 OS/Arch:      linux/amd64
  ```

The project I'm starting codeclimate on: https://github.com/digipolisgent/robo-drupal-console

**Comment by wfleming** [comment]
Hi @daften,

Sorry for the trouble here: our docs were not correct. 

Path is not a required argument for `analyze`, and in normal cases where you want to analyze your entire project probably actually doesn't do what you want. The reason being that providing an explicit path for analysis does not apply any `exclude_paths` you have configured. I've [updated the README & the help output](https://github.com/codeclimate/codeclimate/pull/535) to reflect this.

That's why `analyze .` was taking a long time & using large amounts of CPU but `analyze` did what you expected: it was probably analyzing a large number of files you normally exclude.

Your `docker run ...` command not finding any issues is a more confusing thing. A couple questions there:

1. I notice a different set of engines in the output of the first `docker run ...` you ran vs the output you got from `codecimate analyze`. Did you change your `.codeclimate.yml` in between these commands, or would you have expected the two commands to run the same engines when you were running them?

1. Can you please share the content of your `.codeclimate.yml` and also try running the `docker run` with DEBUG turned on & the `.` arg removed? e.g.

    ```
    docker run \
      --interactive --tty --rm \
      --env CODECLIMATE_DEBUG=1 \
      --env CODECLIMATE_CODE="/code" \
      --volume "$PWD":/code \
      --volume /var/run/docker.sock:/var/run/docker.sock \
      --volume /tmp/cc:/tmp/cc \
      codeclimate/codeclimate analyze
    ```

Thanks.

**Comment by daften** [comment]
No problem, it happens and this is an extremely quick fix. I'm impressed TBH :)

The different engines are because I changed the codeclimate file, there is nothing wrong in that regard. I can rerun with the "big" set of engines if you want, but that's not the problem (I temporarily disabled some engines to see if that would speed things up, it didn't).

Codeclimate file: https://raw.githubusercontent.com/digipolisgent/robo-drupal-console/feature/codeclimate/.codeclimate.yml
Run with DEBUG on: https://gist.github.com/daften/f5351066fde2bf13dd9a9dfe1c47994e

The output about RoboFile.php not existing is what made me try the . at the end of the command.

If you need anything else for debugging, please let me know, and thanks for the help so far!

**Comment by wfleming** [comment]
Thanks for the links. I had clicked through to your project before, but didn't think to look for a branch with the Code Climate config.

I cloned locally & was able to reproduce, and after comparing that docker run command more closely to what our `codeclimate` wrapper script does, I saw the issue (which I should have noticed earlier). the `CODECLIMATE_CODE` env var is expected to be the original host path that the repository lives at (i.e. `$PWD`), not the internal path mounted into the docker container (i.e. `/code`).

The reason that ENV var exists is because the CLI runs each engine in a separate docker container, and when it does it needs to give it a volume mount for the repo code into `/code` for analysis. But, since the CLI is running using your machines's docker daemon, the source path for that path needs to be the real path the code exists at, not the path it exists at inside the CLI container.  With `CODECLIMATE_CODE=/code`, the CLI was trying to run something like `docker run -v /code:/code codeclimate/codeclimate-phpcodesniffer`, when what it should have been running was `docker run -v /your/home/projects/repo:/code codeclimate/codeclimate-phpcodesniffer`.

So the correct manual docker invocation would be

```
docker run \
  --interactive --tty --rm \
  --env CODECLIMATE_CODE="$PWD" \
  --volume "$PWD":/code \
  --volume /var/run/docker.sock:/var/run/docker.sock \
  --volume /tmp/cc:/tmp/cc \
  codeclimate/codeclimate analyze
```

This is another instance where the README is incorrect, so I'll fix that shortly. Thanks for catching all of these issues!
